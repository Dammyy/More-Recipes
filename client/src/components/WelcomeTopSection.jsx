import React, { PureComponent } from 'react';
import { DefaultPlayer as Video } from 'react-html5video';
import 'react-html5video/dist/styles.css';
import PropTypes from 'prop-types';
import ActionButtons from './HomeActionButtons';
/**
 *
 *
 * @class WelcomeTopSetion
 * @extends {PureComponent}
 */
class WelcomeTopSetion extends PureComponent {
  /**
   *
   * @memberOf WelcomeTopSection
   * @returns {void}
   *
   */
  render() {
    const {
      firstName, logout
    } = this.props;
    return (
      <div>
        <div className="col-md-12 welcome-msg-top">
          <i><h1>Welcome to more recipes!</h1></i>
          <div className="intro-msg">
        Find and share everyday cooking inspiration on More Recipes.
          Discover recipes,cooks and how-tos based on the food you love.
          </div>
          <div className="col-md-12 home-logged-in">
            <ActionButtons logout={logout} firstName={firstName} />
          </div>
        </div>
        <div className="container-fluid welcome-section">
          <div className="welcome-video col-md-12">
            <Video
              autoPlay
              loop
              muted
              poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
            >
              <source
                src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                type="video/webm"
              />
            </Video>
          </div>
          <div className="col-md-12 welcome-msg">
            <i><h1>Welcome to more recipes!</h1></i>
            <div className="intro-msg">
              Find and share everyday cooking inspiration on More Recipes.
                Discover recipes,cooks and how-tos based on the food you love.
            </div>
            <div className="col-md-12 home-logged-in">
              <ActionButtons logout={logout} firstName={firstName} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
WelcomeTopSetion.propTypes = {
  firstName: PropTypes.string,
  logout: PropTypes.func.isRequired,
};

WelcomeTopSetion.defaultProps = {
  firstName: 'stuff'
};
export default WelcomeTopSetion;
