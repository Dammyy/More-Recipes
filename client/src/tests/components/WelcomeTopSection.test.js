import React from 'react';
import { shallow } from 'enzyme';
import WelcomeTopSection from '../../components/WelcomeTopSection';

describe('WelcomeTopSection component', () => {
  describe('Snapshot', () => {
    test('component should fully render', () => {
      const props = {
        firstName: '',
        logout: jest.fn(),
      };
      const component = shallow(<WelcomeTopSection {...props} />);
      expect(component).toMatchSnapshot();
    });
  });
});
