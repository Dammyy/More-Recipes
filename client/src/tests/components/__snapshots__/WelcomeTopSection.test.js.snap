// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeTopSection component Snapshot component should fully render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WelcomeTopSetion
    firstName=""
    logout={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="col-md-12 welcome-msg-top"
        >
          <i>
            <h1>
              Welcome to more recipes!
            </h1>
          </i>
          <div
            className="intro-msg"
          >
            Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
          </div>
          <div
            className="col-md-12 home-logged-in"
          >
            <Connect(authAddRecipe(HomeActionButtons))
              firstName=""
              logout={[MockFunction]}
            />
          </div>
        </div>,
        <div
          className="container-fluid welcome-section"
        >
          <div
            className="welcome-video col-md-12"
          >
            <Video
              autoPlay={true}
              loop={true}
              muted={true}
              poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
            >
              <source
                src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                type="video/webm"
              />
            </Video>
          </div>
          <div
            className="col-md-12 welcome-msg"
          >
            <i>
              <h1>
                Welcome to more recipes!
              </h1>
            </i>
            <div
              className="intro-msg"
            >
              Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
            </div>
            <div
              className="col-md-12 home-logged-in"
            >
              <Connect(authAddRecipe(HomeActionButtons))
                firstName=""
                logout={[MockFunction]}
              />
            </div>
          </div>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i>
              <h1>
                Welcome to more recipes!
              </h1>
            </i>,
            <div
              className="intro-msg"
            >
              Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
            </div>,
            <div
              className="col-md-12 home-logged-in"
            >
              <Connect(authAddRecipe(HomeActionButtons))
                firstName=""
                logout={[MockFunction]}
              />
            </div>,
          ],
          "className": "col-md-12 welcome-msg-top",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Welcome to more recipes!
              </h1>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome to more recipes!",
              },
              "ref": null,
              "rendered": "Welcome to more recipes!",
              "type": "h1",
            },
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
              "className": "intro-msg",
            },
            "ref": null,
            "rendered": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Connect(authAddRecipe(HomeActionButtons))
                firstName=""
                logout={[MockFunction]}
              />,
              "className": "col-md-12 home-logged-in",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "firstName": "",
                "logout": [MockFunction],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="welcome-video col-md-12"
            >
              <Video
                autoPlay={true}
                loop={true}
                muted={true}
                poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
              >
                <source
                  src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                  type="video/webm"
                />
              </Video>
            </div>,
            <div
              className="col-md-12 welcome-msg"
            >
              <i>
                <h1>
                  Welcome to more recipes!
                </h1>
              </i>
              <div
                className="intro-msg"
              >
                Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
              </div>
              <div
                className="col-md-12 home-logged-in"
              >
                <Connect(authAddRecipe(HomeActionButtons))
                  firstName=""
                  logout={[MockFunction]}
                />
              </div>
            </div>,
          ],
          "className": "container-fluid welcome-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Video
                autoPlay={true}
                loop={true}
                muted={true}
                poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
              >
                <source
                  src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                  type="video/webm"
                />
              </Video>,
              "className": "welcome-video col-md-12",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoPlay": true,
                "children": <source
                  src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                  type="video/webm"
                />,
                "loop": true,
                "muted": true,
                "poster": "http://wyregate.com/wp-content/uploads/2018/03/img.png",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "src": "http://wyregate.com/wp-content/uploads/2018/03/video.mp4",
                  "type": "video/webm",
                },
                "ref": null,
                "rendered": null,
                "type": "source",
              },
              "type": [Function],
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i>
                  <h1>
                    Welcome to more recipes!
                  </h1>
                </i>,
                <div
                  className="intro-msg"
                >
                  Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
                </div>,
                <div
                  className="col-md-12 home-logged-in"
                >
                  <Connect(authAddRecipe(HomeActionButtons))
                    firstName=""
                    logout={[MockFunction]}
                  />
                </div>,
              ],
              "className": "col-md-12 welcome-msg",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1>
                    Welcome to more recipes!
                  </h1>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Welcome to more recipes!",
                  },
                  "ref": null,
                  "rendered": "Welcome to more recipes!",
                  "type": "h1",
                },
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
                  "className": "intro-msg",
                },
                "ref": null,
                "rendered": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Connect(authAddRecipe(HomeActionButtons))
                    firstName=""
                    logout={[MockFunction]}
                  />,
                  "className": "col-md-12 home-logged-in",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "firstName": "",
                    "logout": [MockFunction],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="col-md-12 welcome-msg-top"
          >
            <i>
              <h1>
                Welcome to more recipes!
              </h1>
            </i>
            <div
              className="intro-msg"
            >
              Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
            </div>
            <div
              className="col-md-12 home-logged-in"
            >
              <Connect(authAddRecipe(HomeActionButtons))
                firstName=""
                logout={[MockFunction]}
              />
            </div>
          </div>,
          <div
            className="container-fluid welcome-section"
          >
            <div
              className="welcome-video col-md-12"
            >
              <Video
                autoPlay={true}
                loop={true}
                muted={true}
                poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
              >
                <source
                  src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                  type="video/webm"
                />
              </Video>
            </div>
            <div
              className="col-md-12 welcome-msg"
            >
              <i>
                <h1>
                  Welcome to more recipes!
                </h1>
              </i>
              <div
                className="intro-msg"
              >
                Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
              </div>
              <div
                className="col-md-12 home-logged-in"
              >
                <Connect(authAddRecipe(HomeActionButtons))
                  firstName=""
                  logout={[MockFunction]}
                />
              </div>
            </div>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i>
                <h1>
                  Welcome to more recipes!
                </h1>
              </i>,
              <div
                className="intro-msg"
              >
                Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
              </div>,
              <div
                className="col-md-12 home-logged-in"
              >
                <Connect(authAddRecipe(HomeActionButtons))
                  firstName=""
                  logout={[MockFunction]}
                />
              </div>,
            ],
            "className": "col-md-12 welcome-msg-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1>
                  Welcome to more recipes!
                </h1>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Welcome to more recipes!",
                },
                "ref": null,
                "rendered": "Welcome to more recipes!",
                "type": "h1",
              },
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
                "className": "intro-msg",
              },
              "ref": null,
              "rendered": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Connect(authAddRecipe(HomeActionButtons))
                  firstName=""
                  logout={[MockFunction]}
                />,
                "className": "col-md-12 home-logged-in",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "firstName": "",
                  "logout": [MockFunction],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="welcome-video col-md-12"
              >
                <Video
                  autoPlay={true}
                  loop={true}
                  muted={true}
                  poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
                >
                  <source
                    src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                    type="video/webm"
                  />
                </Video>
              </div>,
              <div
                className="col-md-12 welcome-msg"
              >
                <i>
                  <h1>
                    Welcome to more recipes!
                  </h1>
                </i>
                <div
                  className="intro-msg"
                >
                  Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
                </div>
                <div
                  className="col-md-12 home-logged-in"
                >
                  <Connect(authAddRecipe(HomeActionButtons))
                    firstName=""
                    logout={[MockFunction]}
                  />
                </div>
              </div>,
            ],
            "className": "container-fluid welcome-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Video
                  autoPlay={true}
                  loop={true}
                  muted={true}
                  poster="http://wyregate.com/wp-content/uploads/2018/03/img.png"
                >
                  <source
                    src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                    type="video/webm"
                  />
                </Video>,
                "className": "welcome-video col-md-12",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "autoPlay": true,
                  "children": <source
                    src="http://wyregate.com/wp-content/uploads/2018/03/video.mp4"
                    type="video/webm"
                  />,
                  "loop": true,
                  "muted": true,
                  "poster": "http://wyregate.com/wp-content/uploads/2018/03/img.png",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "src": "http://wyregate.com/wp-content/uploads/2018/03/video.mp4",
                    "type": "video/webm",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "source",
                },
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i>
                    <h1>
                      Welcome to more recipes!
                    </h1>
                  </i>,
                  <div
                    className="intro-msg"
                  >
                    Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.
                  </div>,
                  <div
                    className="col-md-12 home-logged-in"
                  >
                    <Connect(authAddRecipe(HomeActionButtons))
                      firstName=""
                      logout={[MockFunction]}
                    />
                  </div>,
                ],
                "className": "col-md-12 welcome-msg",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h1>
                      Welcome to more recipes!
                    </h1>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Welcome to more recipes!",
                    },
                    "ref": null,
                    "rendered": "Welcome to more recipes!",
                    "type": "h1",
                  },
                  "type": "i",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
                    "className": "intro-msg",
                  },
                  "ref": null,
                  "rendered": "Find and share everyday cooking inspiration on More Recipes. Discover recipes,cooks and how-tos based on the food you love.",
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Connect(authAddRecipe(HomeActionButtons))
                      firstName=""
                      logout={[MockFunction]}
                    />,
                    "className": "col-md-12 home-logged-in",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "firstName": "",
                      "logout": [MockFunction],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
